src/assimp/assimp.cpp

```diff
@@ -1023,7 +1094,14 @@ vsg::ref_ptr<vsg::Object> assimp::Implementation::read(const vsg::Path& filename
 
             SceneConverter converter;
             converter.filename = filename;
-            return converter.visit(scene, opt, ext);
+
+            if (auto root = converter.visit(scene, opt, ext))
+            {
+                if (converter.externalTextures && converter.externalObjects && !converter.externalObjects->entries.empty())
+                    root->setObject("external", converter.externalObjects);
+
+                return root;
+            }
```

Comment:
The above code block changes what happens when a null ref_ptr is returned from convert,visit(), and would cause the code to fall through to the vsg::read() at the end of method.   Is this intended?


 I would have thought a else return {} would keep the original behavior.

Reply:
Yes you are correct. I mentally completely blanked out the #if .. #endif block and its contained 'return' statement (without that block the behaviour is actually unchanged, but still probably less easy to read).

I hope to get around to updating this PR beginning of next week.

Reply:
I updated the PR now